src/driver/mysql/MysqlDriver.ts

```diff
@@ -530,7 +530,9 @@ export class MysqlDriver implements Driver {
     protected loadDependencies(): void {
         try {
             this.mysql = PlatformTools.load("mysql");  // try to load first supported package
-
+            if (Object.keys(this.mysql).length === 0) {
+                throw new Error("'mysql' was found but it is empty. Falling back to 'mysql2'.");
```

Comment:
The message is there for documentation purposes only. It is never shown to the user since the error gets caught right below.

Reply:
please add detailed comments on this code and explain why you did this and left github issue number.